RecoEgamma/EgammaPhotonProducers/src/ReducedEGProducer.cc

```diff
@@ -943,3 +997,38 @@ void ReducedEGProducer::relinkGsfElectronCore(reco::GsfElectron& gsfElectron,
     gsfElectron.setCore(coreref);
   }
 }
+
+void ReducedEGProducer::calibratePhoton(reco::Photon& photon, 
+					const reco::PhotonRef& oldPhoRef,
+					const edm::ValueMap<float>& energyMap,
+					const edm::ValueMap<float>& energyErrMap)
+{
+  float newEnergy = energyMap[oldPhoRef];
+  float newEnergyErr = energyErrMap[oldPhoRef];
+  photon.setCorrectedEnergy(reco::Photon::P4type::regression2, newEnergy, newEnergyErr, true);
+ 
+}
+
+void ReducedEGProducer::calibrateElectron(reco::GsfElectron& electron,
+					  const reco::GsfElectronRef& oldEleRef,
+					  const edm::ValueMap<float>& energyMap,
+					  const edm::ValueMap<float>& energyErrMap,
+					  const edm::ValueMap<float>& ecalEnergyMap,
+					  const edm::ValueMap<float>& ecalEnergyErrMap)
+{
+  const float newEnergy = energyMap[oldEleRef];
+  const float newEnergyErr = energyErrMap[oldEleRef];
+  const float newEcalEnergy = ecalEnergyMap[oldEleRef];
+  const float newEcalEnergyErr = ecalEnergyErrMap[oldEleRef];
+  
+  electron.setCorrectedEcalEnergy(newEcalEnergy);
+  electron.setCorrectedEcalEnergyError(newEcalEnergyErr);
+ 
+  const math::XYZTLorentzVector oldP4 = electron.p4();
+  math::XYZTLorentzVector newP4 = math::XYZTLorentzVector(oldP4.x() * newEnergy / oldP4.t(),
```

Comment:
please define a temporary to avoid copy-paste. e.g. ```const float corr = newEnergy / oldP4.t(); ``` and then use it in the newP4

Reply:
done